### PR TITLE
[release 1.11] Bump AAQ to v1.1.4, and rename the component name

### PR DESCRIPTION
**Pull request by nunnatsa**
## What this PR does / why we need it
AAQ now called "application-aware-quota". One of the configuration field name also renamed.

This PR changes the import path of AAQ to the new path, bumps it to v1.1.4, and rename the component and the `enableClusterAppsResourceQuota` field to
`allowApplicationAwareClusterResourceQuota`, to match the rename in AAQ.

Also, added a new feature gate, "enableApplicationAwareQuota", to enable/disable AAQ.

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

### Jira Ticket

```jira-ticket
https://issues.redhat.com/browse/CNV-37898
https://issues.redhat.com/browse/CNV-37938
```

**Release note**:

```release-note
Bump AAQ to v1.1.4

Add feature gate for AAQ
```

### CODE REVIEW

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -461,6 +462,12 @@ type HyperConvergedFeatureGates struct {
 	// +kubebuilder:default=false
 	// +default=false
 	AlignCPUs *bool `json:"alignCPUs,omitempty"`
+
+	// EnableApplicationAwareQuota if true, enable the Application Aware Quota feature
```
Comment (orenc1) [inline api/v1beta1/hyperconverged_types.go:466]:
```suggestion
	// EnableApplicationAwareQuota if true, enables the Application Aware Quota feature
```

File: controllers/operands/aaq_test.go [hunk 2]
```diff
@@ -204,7 +204,7 @@ var _ = Describe("AAQ tests", func() {
 		})
 
 		It("should create AAQ if the applicationAwareConfig field exists", func() {
-			hco.Spec.ApplicationAwareConfig = &v1beta1.ApplicationAwareConfigurations{}
+			hco.Spec.FeatureGates.EnableApplicationAwareQuota = ptr.To(true)
```
Comment (orenc1) [inline controllers/operands/aaq_test.go:207]: the test case description doesn't match what we actually checking (`applicationAwareConfig` vs. the FG)

Comment (nunnatsa) [reply controllers/operands/aaq_test.go:207]: Will fixed in future PR

File: api/v1beta1/hyperconverged_types.go (hunk 1 above)
Comment (nunnatsa) [reply api/v1beta1/hyperconverged_types.go:466]: Will fixed in future PR

File: docs/cluster-configuration.md [hunk 3]
```diff
@@ -1117,19 +1125,12 @@ With the `Custom` profile, the cipher list should be expressed according to Open
...
-To enable the AAQ feature, add the `applicationAwareConfig` field to the HyperConverged spec, with an empty object 
-or with custom settings (see below).
-
-For example:
-```yaml
-spec:
-  applicationAwareConfig: {}
-```
+## Configure Application Aware Quota (AAQ)
+To enable the AAQ feature, add the `spec.featureGates.enableApplicationAwareQuota` field to `true`. See [featureGates](#enableapplicationawarequota-feature-gate) above.
```
Comment (orenc1) [inline docs/cluster-configuration.md:1129]:
```suggestion
To enable the AAQ feature, set the `spec.featureGates.enableApplicationAwareQuota` field to `true`. See [featureGates](#enableapplicationawarequota-feature-gate) above.
```

File: docs/cluster-configuration.md [hunk 4]
```diff
@@ -1117,19 +1125,12 @@ With the `Custom` profile, the cipher list should be expressed according to Open
...
 
 To configure AAQ, set the fields of the `applicationAwareConfig` object in the HyperConverged resource's spec. The 
 `applicationAwareConfig` object contains several fields:
-* `vmiCalcConfigName` - determine how resource allocation will be done with ApplicationsResourceQuota.
+* `vmiCalcConfigName` - determine how resource allocation will be done with ApplicationResourceQuota.
```
Comment (orenc1) [inline docs/cluster-configuration.md:1133]:
```suggestion
* `vmiCalcConfigName` - determines how resource allocation will be done with ApplicationResourceQuota.
```

**Review by orenc1 — APPROVED** [review]
please fix the minor typos in a followup PR, lgtm for now
/approve

**Comment by nunnatsa** [comment]
> please fix the minor typos in a followup PR

Done in #2792

**Pull request merged**